vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,308 @@
+        FileUtilities.wipeDirectory(ourInstallerDownloadFolder);
+        const installerPath = path.join(ourInstallerDownloadFolder, `${installerUrl.split('/').slice(-1)}`);
+        await this.download(installerUrl, installerPath);
+        return installerPath;
+    }
+
+    /**
+     *
+     * @returns an empty promise. It will download the file from the url. The url is expected to be a file server that responds with the file directly.
+     * We cannot use a simpler download pattern because we need to download and match the installer file exactly as-is from the server as opposed to writing/copying the bits we are given.
+     */
+    private async download(url : string, dest : string) {
+        return new Promise<void>((resolve, reject) => {
+
+            const installerDir = path.dirname(dest);
+            if (!fs.existsSync(installerDir)){
+                fs.mkdirSync(installerDir);
+            }
+
+            // The file has already been downloaded before. Note that a user couldve added a file here. This is part of why we should sign check the file before launch.
+            if(fs.existsSync(dest))
+            {
+                resolve();
+            }
+
+            const file = fs.createWriteStream(dest, { flags: 'wx' });
+
+            const request = https.get(url, response => {
+                if (response.statusCode === 200)
+                {
+                    response.pipe(file);
+                }
+                else
+                {
+                    file.close();
+                    fs.unlink(dest, () => {}); // Delete incomplete file download
+                    reject(`Server responded with ${response.statusCode}: ${response.statusMessage}`);
+                }
+            });
+
+            request.on('error', err =>
+            {
+                file.close();
+                fs.unlink(dest, () => {}); // Delete incomplete file download
+                reject(err.message);
+            });
+
+            file.on('finish', () =>
+            {
+                resolve();
+            });
+
+            file.on('error', err =>
+            {
+                file.close();
+
+                if (err.message.includes('EEXIST'))
+                {
+                    // 2+ concurrent requests to download the installer occurred and ours got to the race last.
+                    resolve();
+                }
+                else
+                {
+                    fs.unlink(dest, () => {}); // Delete incomplete file download
+                    reject(err.message);
+                }
+            });
+        });
+    }
+
+    public async getExpectedGlobalSDKPath(specificSDKVersionInstalled : string, installedArch : string) : Promise<string>
+    {
+        if(os.platform() === 'win32')
+        {
+            if(installedArch === 'x32')
+            {
+                return path.resolve(`C:\\Program Files (x86)\\dotnet\\sdk\\`);
```

Comment:
You need to resolve these using the native [APIs](https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetspecialfolderpathw)

If you can't invoke the native APIs, you can fall back to environment variables

More importantly, you're assuming this is located on C: drive.

Reply:
Astounding catch, thank you. We can't use that, for the env vars, do you mean `programfiles(x86)` and `programfiles`?

Reply:
I made it use program files, x86 is wrong here